src/app-utils/add-shell-components.tsx

```diff
@@ -111,6 +112,11 @@ export const addComponentsToShell = async (): Promise<void> => {
 		secondaryBar: SidebarView,
 		appView: AppView
 	});
+	upsertApp({
+		name: MAIL_APP_ID,
+		display: label,
+		description: t('label.app_description', 'Mails module')
+	});
```

Comment:
This upsert should be placed as last invocation to make it work also with the shell that still includes the deprecation.